test/extended/router/metrics.go

```diff
@@ -331,48 +328,6 @@ func (t *promTargets) Expect(l promLabels, health, scrapeURLPattern string) erro
 	return fmt.Errorf("no match for %v with health %s and scrape URL %s", l, health, scrapeURLPattern)
 }
 
-func waitForServiceAccountInNamespace(c clientset.Interface, ns, serviceAccountName string, timeout time.Duration) error {
-	w, err := c.CoreV1().ServiceAccounts(ns).Watch(context.Background(), metav1.SingleObject(metav1.ObjectMeta{Name: serviceAccountName}))
-	if err != nil {
-		return err
-	}
```

Comment:
Should we check the SA is present before we attempt to create a token for it?

Reply:
No. There is no longer a need to "wait" for it, and if it does not exist, the error will be descriptive enough (i.e. failed to create token: serviceaccounts "prometheus-k8s" not found).